common/src/main/java/com/wynntils/features/user/overlays/map/MiniMapOverlayFeature.java

```diff
@@ -100,33 +109,130 @@ public void render(PoseStack poseStack, float partialTicks, Window window) {
                     // }
             }
 
-            // enable rotation if necessary
-            if (followPlayerRotation) {
-                poseStack.pushPose();
-                RenderUtils.rotatePose(poseStack, centerX, centerZ, 180 - yRot);
-            }
-
-            renderMapQuad(map, poseStack, centerX, centerZ, textureX, textureZ, width, height);
-
-            // disable rotation if necessary
-            if (followPlayerRotation) {
-                poseStack.popPose();
+            if (WebManager.isMapLoaded()) {
+                renderMapQuad(map, poseStack, centerX, centerZ, textureX, textureZ, width, height);
             }
 
             // TODO minimap icons
 
             // TODO compass icon
 
             // cursor
+            renderCursor(poseStack, centerX, centerZ);
+
+            // disable mask & render border
+            switch (maskType) {
+                case Rectangular -> {
+                    RenderSystem.disableScissor();
+                    renderRectangularMapBorder(poseStack, renderX, renderY, width, height);
+                }
+                    // case Circle -> {
+                    // TODO
+                    // renderCircularMapBorder();
+                    // }
+            }
+
+            // Directional Text
+            renderCardinalDirections(poseStack, width, height, centerX, centerZ);
+
+            // Coordinates
+            if (showCoords) {
+                FontRenderer.getInstance()
+                        .renderText(
+                                poseStack,
+                                String.format(
+                                        "%s, %s, %s",
+                                        (int) McUtils.player().getX(),
+                                        (int) McUtils.player().getY(),
+                                        (int) McUtils.player().getZ()),
+                                centerX,
+                                renderY + height + 6,
+                                CommonColors.WHITE,
+                                HorizontalAlignment.Center,
+                                VerticalAlignment.Middle,
+                                FontRenderer.TextShadow.NONE);
+            }
+        }
+
+        private void renderCardinalDirections(
+                PoseStack poseStack, float width, float height, float centerX, float centerZ) {
+            if (showCompass != CompassRenderType.None) {
```

Comment:
Invert if?

Reply:
I don't understand what you are saying?

Reply:
If renderType is None, return. If I see the code correctly, this would reduce right shift.